###################################
# Collision Check (PlanningScene) #
###################################

# Benchmark configuration (independent of profilers)
benchmark_config:
  parameters:
    runs: 10000                     # Number of trials for each query
    name: "Collision Check (Mesh Quality)"   # Default ""    | Overridden by ROS Param /benchmark/name
    verbose_status_trial: False     # Print status before each trial of all queries
    verbose_status_query: True      # Print status before each query

# Hyperparametization of different robot geometries : approximation using primitive boxes,
# convex mesh approximation generated from detailed meshes and the default hq mesh supplied
# from universal_robot ros package.
#
# Note: Scenes may not be equivalent from one robot representation to another.
profiler_config:
  # MSA (urdf, srdf, kinematics, joint limits)
  robots:
  - name: ur3
    resources:
      urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3.xacro
      srdf: package://moveit_benchmark_suite_resources/robots/ur3/config/ur3.srdf
      kinematics: package://moveit_benchmark_suite_resources/robots/ur3/config/kinematics.yaml
      joint_limits: package://moveit_benchmark_suite_resources/robots/ur3/config/joint_limits.yaml


  # moveit_msgs/RobotState
  robot_states:
  - name: up
    resource:
      joint_state:
        header:
          frame_id: base_link
        name: [shoulder_pan_joint, shoulder_lift_joint, elbow_joint, wrist_1_joint, wrist_2_joint, wrist_3_joint]
        position: [0, -1.5707, 0, -1.5707, 0, 0]
  # moveit_msgs/PlanningScene
  scenes:
  - name: empty
    resource: package://moveit_benchmark_suite_resources/scenes/empty/empty.urdf.xacro
  - name: box-1c
    resource:
      cluttered_scene:
        robot_state: up
        object_in_collision: 1
        object_type: BOX
        dimensions: [1, 1, 1]
        scale: [0.06, 0.2]
        rng_in_collision: 0
  - name: box-1nc
    resource:
      cluttered_scene:
        robot_state: up
        object_no_collision: 1
        object_type: BOX
        dimensions: [1, 1, 1]
        scale: [0.06, 0.2]
        rng_no_collision: 0
  - name: box-100-4c
    resource:
      cluttered_scene:
        robot_state: up
        object_in_collision: 4
        object_no_collision: 96
        object_type: BOX
        dimensions: [1, 1, 1]
        scale: [0.06, 0.2]
        rng_in_collision: 0
        rng_no_collision: 0
  - name: box-100nc
    resource:
      cluttered_scene:
        robot_state: up
        object_no_collision: 100
        object_type: BOX
        dimensions: [1, 1, 1]
        scale: [0.06, 0.2]
        rng_no_collision: 0

  # collision_detection::CollisionPlugin class name
  collision_detectors:
  - FCL
  - Bullet

  # collision_detection::CollisionRequest
  collision_requests:
  - name: First (Binary)
    resource:
      distance: False             # Compute proximity distance (Slow)
      cost: False                 # Compute collision cost
      contacts: False             # Compute contacts, otherwise only a binary collision yes/no is reported
      max_contacts: 1             # Maximum number of contacts
      max_contacts_per_pair: 1    # Maximum numbe of contacts per pair of bodies
      max_cost_sources: 1         # Defines how many of the top cost sources should be returned
      verbose: False              # Report information about collision
  - name: First (Contact Only)
    resource:
      distance: False
      cost: False
      contacts: True
      max_contacts: 1
      max_contacts_per_pair: 1
      max_cost_sources: 1
      verbose: False
  - name: First (Cost)
    resource:
      distance: False
      cost: True
      contacts: True
      max_contacts: 1
      max_contacts_per_pair: 1
      max_cost_sources: 1
      verbose: False
  # - name: First (Distance)
  #   resource:
  #     distance: True
  #     cost: False
  #     contacts: True
  #     max_contacts: 1
  #     max_contacts_per_pair: 1
  #     max_cost_sources: 1
  #     verbose: False
  - name: All (Contact Only)
    resource:
      distance: False
      cost: False
      contacts: True
      max_contacts: 99
      max_contacts_per_pair: 1
      max_cost_sources: 1
      verbose: False
  - name: All (Cost)
    resource:
      distance: False
      cost: True
      contacts: True
      max_contacts: 99
      max_contacts_per_pair: 1
      max_cost_sources: 99
      verbose: False
  # - name: All (Distance)
  #   resource:
  #     distance: True
  #     cost: False
  #     contacts: True
  #     max_contacts: 99
  #     max_contacts_per_pair: 1
  #     max_cost_sources: 1
  #     verbose: False

extend_resource_config:
  robots:
    extend_resource: ur3
    resource_names:
      - primitive_boxes
      - vhacd_h01
      - vhacd_h02
      - vhacd_h03
      - vhacd_h04
      - vhacd_h05
      - vhacd_h06
      - vhacd_h07
      - vhacd_h08
      - vhacd_h09
      - vhacd_h10
      - vhacd_h11
      - vhacd_h12
      - vhacd_h13
      - vhacd_h14
      - vhacd_h15
      - vhacd_h16
      - default_mesh
    resources:
      # lowest to highest quality of robot representation
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_bbox.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h01.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h02.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h03.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h04.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h05.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h06.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h07.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h08.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h09.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h10.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h11.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h12.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h13.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h14.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h15.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3_vhacd_h16.xacro
      - urdf: package://moveit_benchmark_suite_resources/robots/ur3/urdf/ur3.xacro
